### PR TITLE
[AppConfig] `az appconfig kv export/import`: Add new parameters to support exporting and importing all key values from a snapshot of the source configuration

### PR DESCRIPTION
**Pull request by albertofori**
**Related command**

`az appconfig kv export`
`az appconfig kv import`

**Description**

This change enables users to export key-values from a configuration snapshot to other destinations.

**Testing Guide**

**History Notes**

[AppConfig] `az appconfig kv export/import`: Add a new parameter `--snapshot` to support exporting and importing all key values from a snapshot of the source configuration

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppConfig

File: src/azure-cli/azure/cli/command_modules/appconfig/_params.py [hunk 1]
```diff
@@ -217,6 +217,7 @@ def load_arguments(self, _):
         c.argument('yes', help="Do not prompt for preview.")
         c.argument('skip_features', help="Export items excluding all feature flags. By default, all features with the specified label will be exported to file or appconfig. Not applicable for appservice.", arg_type=get_three_state_flag())
         c.argument('skip_keyvault', help="Export items excluding all key vault references. By default, all key vault references with the specified label will be exported.", arg_type=get_three_state_flag())
+        c.argument('snapshot', help="List all keys in a given snapshot of the App Configuration store. If no snapshot is specified, the keys currently in the store are listed.", is_preview=True)
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/_params.py:220]:
```suggestion
        c.argument('snapshot', help="Export all keys in a given snapshot of the App Configuration store. If no snapshot is specified, the keys currently in the store are exported.", is_preview=True)
```

Comment (avanigupta) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py:220]: Is that true though? If no snapshot specified, we'll export by matching key and label filter (not all keys in the store), right?

Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py:220]: That's right. How about this?
`Export all keys in a given snapshot of the App Configuration store. If no snapshot is specified, the keys currently in the store are exported based on the specified key and label filters.`

Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/_params.py]: This has been updated, thanks!

**Review by avanigupta — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
Could you please add a test case for this new parameter?

**Comment by albertofori** [comment]
> Could you please add a test case for this new parameter?
@zhoxing-ms Please the test cases have been updated to test the 2 new parameters.

**Comment by albertofori** [comment]
> > Could you please add a test case for this new parameter?
@zhoxing-ms Please the test cases have been updated to test the new parameters.

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@albertofori Please resolve this conflict

**Comment by albertofori** [comment]
> @albertofori Please resolve this conflict

@zhoxing-ms Please conflicts have been resolved, however, could you hold on from merging for now as this needs another review for functionality that was added for import.

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py [hunk 2]
```diff
@@ -123,6 +128,7 @@ def import_config(cmd,
 
         src_kvs = __read_kv_from_config_store(src_azconfig_client,
                                               key=src_key,
+                                              snapshot=src_snapshot,
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:131]: Do we need to read feature flags from this snapshot?

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py [hunk 3]
```diff
@@ -67,11 +67,16 @@ def import_config(cmd,
...
                   src_label=None,
+                  src_snapshot=None,
                   preserve_labels=False,
                   src_auth_mode="key",
                   src_endpoint=None,
                   # from-appservice parameters
                   appservice_account=None):
+
+    if src_snapshot and (src_key or src_label):
+        raise CLIErrors.MutuallyExclusiveArgumentError("'snapshot' cannot be specified with 'key' or 'label' filters.")
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:78]:
```suggestion
        raise CLIErrors.MutuallyExclusiveArgumentError("'snapshot' cannot be specified with '--src-key' or '--src-label' filters.")
```

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py (hunk 2 above)
Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:131]: You are right. I missed this.  For snapshots we don't filter out any kvs. When a user specifies `--skip-keyvault` or `--skip-features` with snapshots, should we fail the command or just log a warning and import all kvs?

Comment (avanigupta) [reply src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:131]: I think we should throw an error. When importing/exporting from a snapshot, we dont allow any further filtering. 
@jimmyca15, thoughts?

Comment (jimmyca15) [reply src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:131]: Agree on throwing an error.

Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:132]: This has been updated accordingly

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py [hunk 4]
```diff
@@ -67,11 +68,16 @@ def import_config(cmd,
...
                   src_label=None,
+                  src_snapshot=None,
                   preserve_labels=False,
                   src_auth_mode="key",
                   src_endpoint=None,
                   # from-appservice parameters
                   appservice_account=None):
+
+    if src_snapshot and (src_key or src_label or skip_features):
+        raise CLIErrors.MutuallyExclusiveArgumentError("'--src-snapshot' cannot be specified with '--src-key', '--src-label' filters, or '--skip-features' options.")
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:79]: Can we do this validation with the argument validators?

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py [hunk 5]
```diff
@@ -250,6 +263,10 @@ def export_config(cmd,
                   # to-app-service parameters
                   appservice_account=None,
                   export_as_reference=False):
+
+    if snapshot and (key or label or skip_features or skip_keyvault):
+        raise CLIErrors.MutuallyExclusiveArgumentError("'--snapshot' cannot be specified with '--key',  '--label', 'skip-keyvault' or '--skip-features' arguments.")
```
Comment (avanigupta) [inline src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py:268]: Same here, if possible, lets move this with the other validators. Also, there's a typo: `--skip-keyvault`

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py (hunk 4 above)
Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py]: That's a good idea. This has been moved. Thanks!

File: src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py (hunk 5 above)
Comment (albertofori) [reply src/azure-cli/azure/cli/command_modules/appconfig/keyvalue.py]: This has been updated, thanks!

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**